cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -0,0 +1,177 @@
+                throw IllegalArgumentException("Borrower must start the flow")
+            }
+            val signedTx = serviceHub.signInitialTransaction(txBuilder)
+
+            progressTracker.currentStep = INITIALISE_OTHER_PARTY_FLOW
+            // Send the state to the counterparty, and receive it back with their signature.
+            val lenderPartySession = initiateFlow(iouState.lender)
+
+            // send/receive part is just for experiment
+            val confirmPayback = lenderPartySession
+                    .sendAndReceive<Boolean>("ready for receiving payback?")
+                    .unwrap { it }
+
+            require(confirmPayback) { "Oops, lender failed to be ready" }
+
+            progressTracker.currentStep = PAYBACK_COMMUNICATION
+            lenderPartySession.send(iouState.value)
+
+            progressTracker.currentStep = GATHERING_SIGS
+            val fullySignedTx = subFlow(CollectSignaturesFlow(
+                    signedTx,
+                    setOf(lenderPartySession),
+                    GATHERING_SIGS.childProgressTracker())
+            )
+
+            progressTracker.currentStep = FINALISING_TRANSACTION
+            // Notarise and record the transaction in both parties' vaults.
+            return subFlow(FinalityFlow(fullySignedTx, setOf(lenderPartySession), FINALISING_TRANSACTION.childProgressTracker()))
+        }
+    }
+
+    @InitiatedBy(Initiator::class)
+    class Acceptor(val borrowerPartySession: FlowSession) : FlowLogic<SignedTransaction>() {
+        @Suspendable
+        override fun call(): SignedTransaction {
+            borrowerPartySession.receive<String>().unwrap { it }
+            //send confirmation about being ready to receive payback
+            borrowerPartySession.send(true)
+
+            // just exercise
+            val receivedPayback: Int = borrowerPartySession.receive<Int>().unwrap { it }
+
+            val signTransactionFlow = object : SignTransactionFlow(borrowerPartySession) {
+                override fun checkTransaction(stx: SignedTransaction) = requireThat {
+                    val iouState = serviceHub.toStateAndRef<IOUState>(stx.tx.inputs.first()).state.data
+                    // val inCashStates = serviceHub.toStateAndRef<CashState>(stx.tx.inputs.last()).state.data
+                    val outCashStates = stx.tx.outputsOfType<CashState>()
+                    val sum = outCashStates.sumBy { it.value.toInt() }
+
+                    "Not enough money to payback. $sum < ${iouState.value}" using (iouState.value.compareTo(sum) <= 0)
+                    "Lender should own IOU." using iouState.lender.owningKey.equals(ourIdentity.owningKey)
+                    "Lender should own payback cash." using outCashStates.all { it.owner == ourIdentity }
+
+                    val bank = serviceHub.bankProvider()
+                    "Accept money created by bank." using outCashStates.all { it.creator == bank }
```

Comment:
Mmh, what if the `creator` is mentioned on the IOU so that there is no need to call to the `bankProvider`.

Reply:
if we add bank to the IOU and bank change in a meantime, then we will stuck. so isn't that better if we check for "current" bank? Or maybe there will be 2 keys/banks that can create money.

Reply:
The (bank) creator of the cash is a `Party`, with a public key. Whether the private is lost or not, owners can still exchange the cash and it does not lose its value. The bank is not involved in the IOU either.
Plus, this consideration of losing the key, is valid for all public keys. Why not the borrower and lender?